lms/static/scripts/frontend_apps/components/dashboard/SyncGradesButton.tsx

```diff
@@ -70,7 +88,17 @@ export default function SyncGradesButton({
       return (
         <>
           Syncing grades
-          <SpinnerCircleIcon className="ml-1.5" />
+          <div
+            className={classnames(
+              'border-solid border-l border-grey-5',
+              // Compensate the button's padding, so that this grows from top to
+              // bottom
+              '-my-2 self-stretch',
+            )}
+          />
+          <div className="text-grey-3 text-[0.7rem] self-center">
+            {syncedStudentsCount}/{totalStudentsBeingSynced}
```

Comment:
`totalStudentsBeingSynced` is potentially undefined here, according to the styles at least. Can that situation actually happen? Looking at this component in isolation it is possible.

Reply:
It should not happen in this particular `if` statement, but it's not easy to define at the type definition level.

I also considered adding an extra `?? 0` fallback in the memoized value, but we would not be able to test it anyway, precisely because when the value is rendered here, it always has a value.

Reply:
> It should not happen in this particular if statement, but it's not easy to define at the type definition level.

Why is that? Are we assuming that `lastSync.data` will always be non-null for values of `lastSync` passed into this component?

Reply:
> Are we assuming that `lastSync.data` will always be non-null for values of `lastSync` passed into this component?

No, but we are explicitly checking for that in order to render this particular block.

```tsx
if (
  lastSync.data &&
  ['scheduled', 'in_progress'].includes(lastSync.data.status)
) {
  // ...
}
```

So when `totalStudentsBeingSynced` is rendered inside that if, we know `lastSync.data.grades.length` is defined, even if in the `useMemo` we have to do the null check.

Reply:
I could inline `{totalStudentsToSync ?? lastSync.data.grades.length}` here instead, and drop `totalStudentsBeingSynced`entirely, but since I had to explain why we were falling back to `lastSync.data.grades.length`, I thought extracting it to a variable would make it easier.

Reply:
OK, well along those lines, you can do this:

```diff
diff --git a/lms/static/scripts/frontend_apps/components/dashboard/SyncGradesButton.tsx b/lms/static/scripts/frontend_apps/components/dashboard/SyncGradesButton.tsx
index 85af7a2e3..afe6c09f5 100644
--- a/lms/static/scripts/frontend_apps/components/dashboard/SyncGradesButton.tsx
+++ b/lms/static/scripts/frontend_apps/components/dashboard/SyncGradesButton.tsx
@@ -61,13 +61,6 @@ export default function SyncGradesButton({
     [lastSync.data?.grades],
   );
   const [totalStudentsToSync, setTotalStudentsToSync] = useState<number>();
-  const totalStudentsBeingSynced = useMemo(
-    // Use the amount set when a sync is started, but fall back to the amount of
-    // students from last sync, in case a sync was in progress when this was
-    // loaded
-    () => totalStudentsToSync ?? lastSync.data?.grades.length,
-    [lastSync.data?.grades.length, totalStudentsToSync],
-  );
   const startSync = useCallback(() => {
     // Right before starting, set amount of students that are going to be synced.
     // This will prevent displaying a zero in the short interval between the
@@ -85,6 +78,12 @@ export default function SyncGradesButton({
       lastSync.data &&
       ['scheduled', 'in_progress'].includes(lastSync.data.status)
     ) {
+      // Use the amount set when a sync is started, but fall back to the amount of
+      // students from last sync, in case a sync was in progress when this was
+      // loaded
+      const totalStudentsBeingSynced =
+        totalStudentsToSync ?? lastSync.data.grades.length;
+
       return (
         <>
           Syncing grades
@@ -137,7 +136,7 @@ export default function SyncGradesButton({
     lastSync.data,
     lastSync.error,
     syncedStudentsCount,
-    totalStudentsBeingSynced,
+    totalStudentsToSync,
   ]);
 
   const buttonDisabled =
```

Reply:
Right! That's a good idea.